fed/config.py

```diff
@@ -80,14 +81,105 @@ def get_job_config():
     return _job_config
 
 
-class ProxyActorConfig:
+class CrossSiloCommConfig:
     """A class to store parameters used for Proxy Actor
 
     Attributes:
-        resource_label: The customized resources for the actor. This will be
-            filled into the "resource" field of Ray ActorClass.options.
+        proxier_fo_max_retries: The max restart times for the send proxy.
+        serializing_allowed_list: The package or class list allowed for
+            serializing(deserializating) cross silos. It's used for avoiding pickle
+            deserializing execution attack when crossing solis.
+        send_resource_label: Customized resource label, the SendProxyActor
+            will be scheduled based on the declared resource label. For example,
+            when setting to `{"my_label": 1}`, then the SendProxyActor will be started
+            only on Nodes with `{"resource": {"my_label": $NUM}}` where $NUM >= 1.
+        recv_resource_label: Customized resource label, the RecverProxyActor
+            will be scheduled based on the declared resource label. For example,
+            when setting to `{"my_label": 1}`, then the RecverProxyActor will be started
+            only on Nodes with `{"resource": {"my_label": $NUM}}` where $NUM >= 1.
+        exit_on_sending_failure: whether exit when failure on
+            cross-silo sending. If True, a SIGTERM will be signaled to self
+            if failed to sending cross-silo data.
+        messages_max_size_in_bytes: The maximum length in bytes of
```

Comment:
`messages_max_size_in_bytes` may be correlated to grpc only. There is no similar param in brpc. So we can move this param to grpc config.

Reply:
brpc has a `max_body_size` to limit the message size,  doc:
https://github.com/apache/brpc/blob/master/docs/en/server.md#limit-sizes-of-messages

I think it takes the same effect ?

Reply:
You are right.
But i still tend to configure max_message_size in their respective configurations, so that we don't have to translate max_message_size to various specific implementations.

E.g., for grpc, users can config in GrpcConfig,
```python
[('grpc.max_send_message_length', 1024), ('grpc.max_receive_message_length', 1024)]
```

for brpc, users can config in BrpcConfig as well.

For users, it is closer to native usage.

Reply:
The conversion always happens when we try to define some common configs.
For example, the `timeout_ms` is definitely a `CrossSiloCommConfig`, but it needs to be translated to the corresponding name or parameter in a certain implementation.

One duty of the proxy implementation is to properly parse the config and set the connection.

I think the key is the semantics of the config, instead the actual name.

Reply:
What i am worried is that some implementations do not have this parameter configuration, but users may mistakenly think that this parameter works.
And another scenario is that user configure `grpc.max_send_message_length` in `grpc_channel_options` at the same time, then which param should take effect?

Reply:
> What i am worried is that some implementations do not have this parameter configuration, but users may mistakenly think that this parameter works. And another scenario is that user configure `grpc.max_send_message_length` in `grpc_channel_options` at the same time, then which param should take effect?

Yes, your worries are reasonable. I've also considered the two scenarios.

But it's hard to define a general config set, because the more general it is, the fewer configs it can contain. Meanwhile, I think the config conflict is quite common to see.

For now, I've put  `messages_max_size_in_bytes` to the grpc section to solve the conflict.

Reply:
> because the more general it is, the fewer configs it can contain

Yes, adding common configuration requires careful consideration. Adding is easy, removing is hard,  we can add later when we feel the need.